lib/parser/livesplit_core_parser.rb

```diff
@@ -62,95 +75,105 @@ def self.parse(run, fast: false)
-          gametime_duration_ms: time.game_time.try(:total_seconds).try(:*, 1000) || 0,
-
-          started_at: attempt_started && DateTime.parse(attempt_started.to_rfc3339),
-          ended_at:   attempt_ended   && DateTime.parse(attempt_ended.to_rfc3339)
-        }
+    run_object[:history] = []
+    (0...run.attempt_history_len).each do |i|
+      attempt = run.attempt_history_index(i)
+      attempt_id = attempt.index.to_i
+      time = attempt.time
 
-        # See https://github.com/glacials/splits-io/pull/474/files#r241242051
-        [attempt_started, attempt_ended].compact.each(&:dispose)
+      # See https://github.com/glacials/splits-io/pull/474/files#r241242051
+      attempt_started = attempt.started
+      attempt_ended = attempt.ended
 
-        if time.real_time.try(:total_seconds).present?
-          run_object[:indexed_history][attempt_id] = time.real_time.try(:total_seconds)
-          run_object[:realtime_history] << time.real_time.try(:total_seconds)
-        end
-      end
-    end
+      run_object[:history] << {
+        attempt_number:       attempt_id,
+        realtime_duration_ms: time.real_time.try(:total_seconds).try(:*, 1000) || 0,
+        gametime_duration_ms: time.game_time.try(:total_seconds).try(:*, 1000) || 0,
 
-    (0...run.len).each do |i|
-      segment = run.segment(i)
-      split = Split.new
-      split.name = segment.name
+        started_at: attempt_started && DateTime.parse(attempt_started.to_rfc3339),
+        ended_at:   attempt_ended   && DateTime.parse(attempt_ended.to_rfc3339),
 
-      split.realtime_end = segment.personal_best_split_time.real_time.try(:total_seconds) || 0
-      split.realtime_duration = [0, split.realtime_end - run_object[:realtime_time]].max
-      split.realtime_start = split.realtime_end - split.realtime_duration
+        pause_duration_ms: attempt.pause_time.try(:total_seconds).try(:*, 1000)
+      }
 
-      split.realtime_best = segment.best_segment_time.real_time.try(:total_seconds) || 0
-      split.realtime_skipped = split.realtime_duration.zero?
+      # See https://github.com/glacials/splits-io/pull/474/files#r241242051
+      [attempt_started, attempt_ended].compact.each(&:dispose)
+    end
 
-      if split.realtime_duration.positive? && split.realtime_duration.round(5) <= split.realtime_best.try(:round, 5)
-        split.realtime_gold = true
-      else
-        split.realtime_gold = false
+    (0...run.len).each do |i|
+      segment = run.segment(i)
+      split = {
+        segment_number: i,
+        name: segment.name,
+        icon: segment.icon.presence,
+        realtime_gold: false,
+        realtime_skipped: false,
+        gametime_gold: false,
+        gametime_skipped: false,
+        history: []
+      }
+
+      split[:realtime_end_ms] = (segment.personal_best_split_time.real_time.try(:total_seconds) || 0) * 1000
+      split[:realtime_duration_ms] = [0, split[:realtime_end_ms] - run_object[:realtime_duration_ms]].max
```

Comment:
I really feel like Github is deleting my comments now since this is more than once now it's happened.

At the end of each split iteration, it adds the current split duration to the running counter to get the actual duration of the run (as opposed to just checking the last split end time), which ends up being the same as the last splits end.  This just means we don't have to make a special case for the first iteration.

Reply:
Ohhh ok, so `run_object[:realtime_duration_ms]` is the run duration _so far_, at the end of the last split. Got it! Also didn't realize before that this is basically unchanged from previous behavior 👍